### PR TITLE
feat(frontend): implement navbar 

### PR DESCRIPTION
**Pull request by Eddyyyh**

### CODE REVIEW

File: frontend/styles/Navbar.module.css [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+.menuContent {
+  position: absolute;
+  width: 100%;
+  background-color: var(--light-yellow);
+  border-radius: 0px 0px 15px 15px;
+  transition: background-color 0.2s, border-radius 0.2s, visibility 0.2s;
+}
+
+.clearMenuContent {
+  background-color: rgba(255, 255, 255, 0.2) !important;
```
Comment (joanna209) [inline frontend/styles/Navbar.module.css:274]: Is there a reason why we're using `!important` in both CSS files?

File: frontend/components/CompactNavigationBar.js [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+  });
+
+  const checkBottomScreen = () => {
+    const checkBottom =
+      Math.ceil(window.innerHeight + window.scrollY) >=
+      document.documentElement.scrollHeight;
+    setHiddenNavBar(checkBottom);
+  };
+
+  const checkTopScreen = () => {
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:76]: Move the logic common in both `CompactNavigationBar` and `NavigationBar` to a new file and import them in. Seems like the logic is repeated (e.g. `checkTopScreen`, `changeAboutUsDropDown`)

File: frontend/components/CompactNavigationBar.js [hunk 3]
```diff
@@ -0,0 +1,440 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Image from "next/image";
+import styles from "../styles/Navbar.module.css";
+
+// import menu icons
+import MenuIcon from "@material-ui/icons/Menu";
+import CloseIcon from "@material-ui/icons/Close";
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:9]: NIT: should be a space between lines 9 and 10

File: frontend/components/CompactNavigationBar.js [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+  const [clearNavBar, setClearNavBar] = useState(true);
+  const [hiddenNavBar, setHiddenNavBar] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [aboutUsDropdownOpen, setAboutUsDropdownOpen] = useState(false);
+  const [mediaDropdownOpen, setMediaDropdownOpen] = useState(false);
+  const [darkMode, setDarkMode] = useState(false);
+  const router = useRouter();
+  const pageLoading = false;
+
+  /* check whether device is in light or dark mode
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:31]: NIT: comments throughout this PR should be one liners i.e. using `//` syntax rather than `/* */` . If we require using the latter syntax, think about whether the comment itself could be simplified.

Also consider whether it is necessary to leave comments. Sometimes if we have good variable names, it should be clear as to what the code is trying to achieve.

File: frontend/pages/index.js [hunk 5]
```diff
@@ -82,7 +82,7 @@ const Home = () => {
 
           {/* Start of Description */}
           <div
-            data-aos={isMobile ? "fade" : "fade-right"}
+            data-aos={isMobile ? "fade" : "fade-up"}
```
Comment (joanna209) [inline frontend/pages/index.js:85]: Is there a reason why these changes are in this PR?

File: frontend/components/CompactNavigationBar.js [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+              aboutUsDropdownOpen
+                ? styles.menuDropdownContent
+                : `${styles.menuDropdownContent} ${styles.hiddenMenuDropdownContent}`
+            }
+          >
+            <Link href="/about/our-story">
+              {/* make the current page name in the dropdown bold,
+                  same structure for the other dropdown links */}
+              <div
+                className={
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:249]: Use the `map` function to iterate through the details of the dropdown rather than having it hardcoded throughout. Take a look at Footer.js for reference.

File: frontend/components/NavigationBar.js [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+  useEffect(() => {
+    let aboutUsDropdown = document.getElementById("aboutUsDropdown");
+    let mediaDropdown = document.getElementById("mediaDropdown");
+    aboutUsDropdown.addEventListener("mouseover", changeAboutUsToArrowDown);
+    aboutUsDropdown.addEventListener("mouseleave", changeAboutUsToArrowRight);
+    mediaDropdown.addEventListener("mouseover", changeMediaToArrowDown);
+    mediaDropdown.addEventListener("mouseleave", changeMediaToArrowRight);
+  }, []);
+
+  const checkWindowWidth = () => {
```
Comment (joanna209) [inline frontend/components/NavigationBar.js:76]: Consider using `{ isMobile }` from `react-device-detect` instead of manually checking the window width

File: frontend/components/NavigationBar.js [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+                height={35}
+              />
+            </a>
+          </Link>
+        </div>
+        <div className={styles.emptyGridItem}></div>
+        <div className={`${styles.homeGridItem} ${styles.linkContainer}`}>
+          <Link href="/">
+            <div className={styles.linkContent}>
+              <a>Home</a>
```
Comment (joanna209) [inline frontend/components/NavigationBar.js:153]: Use the `map` function for routes/headings that aren't included in the about us/media dropdowns too.

File: frontend/styles/Navbar.module.css [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+  font-weight: bold;
+}
+
+.icon {
+  margin-left: -10px;
+  margin-right: 10px;
+  transform: scale(0.9);
+}
+
+/* compact navbar */
```
Comment (joanna209) [inline frontend/styles/Navbar.module.css:207]: Follow the CSS commenting convention for bigger comments (take a look at `global.css` for reference)

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: frontend/components/CompactNavigationBar.js [hunk 10]
```diff
@@ -68,21 +83,59 @@ const CompactNavigationBar = () => {
   };
 
   const menuItemClick = () => {
-    setAboutUsMenuOpen(false);
-    setMediaMenuOpen(false);
+    setAboutUsDropdownOpen(false);
+    setMediaDropdownOpen(false);
     setMenuOpen(false);
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:88]: For the menu item icon (e.g. the three lines and the cross) specifically, can we make these icons white? Red is reserved for links

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: frontend/pages/index.js (hunk 5 above)
Comment (katejoh) [reply frontend/pages/index.js:85]: Using `"fade-left"` creates extra space on the right of the screen until the animation runs, which causes the navbar to extend past the edge of the window - see the screenshot below reference.

image

File: frontend/styles/Navbar.module.css (hunk 1 above)
Comment (Eddyyyh) [reply frontend/styles/Navbar.module.css:274]: We used !important only to change the background colour of the dropdowns and text hover colour for dropdown container. Without using !important, we couldn’t find a way to make the background colour transparent or clear white colour (in compact navbar) when the scroll is at the top of the page.

Comment (joanna209) [reply frontend/styles/Navbar.module.css:274]: Thank you! Epic explanation 👍

File: frontend/pages/index.js (hunk 5 above)
Comment (joanna209) [reply frontend/pages/index.js:85]: Yep sounds good - feel free to leave it in then

File: frontend/components/CompactNavigationBar.js [hunk 11]
```diff
@@ -3,20 +3,19 @@ import { useRouter } from "next/router";
 import Link from "next/link";
 import Image from "next/image";
 import styles from "../styles/Navbar.module.css";
-
-// import menu icons
+import {
+  navigationBarContent,
+  aboutUsDropdownContent,
+  mediaDropdownContent,
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:9]: Remove `aboutUsDropDownContent` and `mediaDropdownContent`, seems like they aren't being used

File: frontend/components/CompactNavigationBar.js [hunk 12]
```diff
@@ -3,20 +3,19 @@ import { useRouter } from "next/router";
...
+} from "../data/navbarData";
+import {
+  changeAboutUsToArrowDown,
+  changeAboutUsToArrowRight,
+  changeMediaToArrowDown,
+  changeMediaToArrowRight,
+} from "./navbarHelpers";
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:16]: Rename the file to `navbar.js` and move the file to the helpers folder (merge develop into this branch)

File: frontend/components/CompactNavigationBar.js [hunk 13]
```diff
@@ -177,260 +147,134 @@ const CompactNavigationBar = () => {
...
+                    >
+                      <a id={page.text} key={index}>
+                        {page.pageName}
+                      </a>
+                    </div>
+                  </div>
+                  <div
+                    className={
+                      page.id === "aboutUsDropdown"
+                        ? aboutUsDropdownOpen
```
Comment (joanna209) [inline frontend/components/CompactNavigationBar.js:213]: Look into making lines 188-205 more readable/maintainable. Perhaps we can use if statements here instead of ternary.

File: frontend/data/navbarData.js [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+import ChromeReaderModeOutlinedIcon from "@material-ui/icons/ChromeReaderModeOutlined";
+import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
+import PeopleOutlineIcon from "@material-ui/icons/PeopleOutline";
+import PhoneOutlinedIcon from "@material-ui/icons/PhoneOutlined";
+import InsertDriveFileOutlinedIcon from "@material-ui/icons/InsertDriveFileOutlined";
+import HeadsetMicOutlinedIcon from "@material-ui/icons/HeadsetMicOutlined";
+import LocalPrintshopOutlinedIcon from "@material-ui/icons/LocalPrintshopOutlined";
+import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
+import VideocamOutlinedIcon from "@material-ui/icons/VideocamOutlined";
+
```
Comment (joanna209) [inline frontend/data/navbarData.js:11]: Rename this file to `NavbarData.js`

File: frontend/styles/Navbar.module.css [hunk 15]
```diff
@@ -352,7 +351,7 @@
 }
 
 
-/*--- Dark mode ---*/
+/*---------------------------------Dark Mode----------------------------------*/
```
Comment (joanna209) [inline frontend/styles/Navbar.module.css:354]: Change this back to the regular comment style

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: frontend/components/CompactNavigationBar.js (hunk 12 above)
Comment (Eddyyyh) [reply frontend/components/CompactNavigationBar.js:16]: donee !

File: frontend/data/navbarData.js (hunk 14 above)
Comment (Eddyyyh) [reply frontend/data/navbarData.js:11]: donee !

**Review by joanna209 — CHANGES_REQUESTED** [review]
@Eddyyyh @katejoh please resolve the failing tests first before requesting review

**Review by joanna209 — APPROVED** [review]
